crates/core/src/client/mod.rs

```diff
@@ -395,25 +363,9 @@ impl<P: Provider + Send + Sync> KakarotEthApi<P> for KakarotClient<P> {
         let starknet_block_id: StarknetBlockId = EthBlockId::new(block_id).try_into()?;
         let starknet_address = self.compute_starknet_address(ethereum_address, &starknet_block_id).await?;
 
-        let request = FunctionCall {
-            // This FieldElement::from_hex_be cannot fail as the value is a constant
-            contract_address: FieldElement::from_hex_be(STARKNET_NATIVE_TOKEN).unwrap(),
-            entry_point_selector: BALANCE_OF,
-            calldata: vec![starknet_address],
-        };
-
-        let balance = self.starknet_provider.call(request, starknet_block_id).await?;
-
-        let balance = balance
-            .first()
-            .ok_or_else(|| DataDecodingError::InvalidReturnArrayLength {
-                entrypoint: "balance".into(),
-                expected: 1,
-                actual: 0,
-            })?
-            .to_bytes_be();
-
-        let balance = U256::from_be_bytes(balance);
+        let balance: Felt252Wrapper =
+            self.kakarot_contract.balance(&self.starknet_provider, &starknet_address, &starknet_block_id).await?.into();
```

Comment:
like for the `AccountContract`, this doesn't really belong to Kakarot, but we definitely need something like a `FeeTokenContract` or just `EthContract`.

Reply:
issue #289 created for this